manage-operational-layers/src/main/java/com/esri/arcgismaps/sample/manageoperationallayers/components/ComposeMapView.kt

```diff
@@ -0,0 +1,72 @@
+/* Copyright 2023 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.manageoperationallayers.components
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.lifecycle.LifecycleOwner
+import com.arcgismaps.mapping.view.MapView
+
+/**
+ * Wraps the MapView in a Composable function.
+ */
+@Composable
+fun ComposeMapView(
+    modifier: Modifier = Modifier,
+    mapViewModel: MapViewModel
+) {
+    // get an instance of the current lifecycle owner
+    val lifecycleOwner = LocalLifecycleOwner.current
+    // get the instance of the MapView state
+    val mapViewState = mapViewModel.mapViewState
+    // create and add MapView to the activity lifecycle
+    val mapView = createMapViewInstance(lifecycleOwner)
+
+    // wrap the MapView as an AndroidView
+    AndroidView(
+        modifier = modifier,
+        factory = { mapView },
```

Comment:
Does it matter that this doesn't create a new `mapView` every time it's called? I would have thought the call to `createMapViewInstance` should go directly in here but maybe that's not necessary when using MapView?

Reply:
We need to create the `mapView` instance from a composable function. The `factory = { }` property of `AndroidView` does not provide a composable context to call the fun `createMapViewInstance`. Which is why it's called before the `AndroidView` composition.